File: programmers/pg_lv2_프로세스.js
function solution(priorities, location) {
  const queue = [];
  priorities.forEach((ele, idx) => {
    queue.push([ele, idx]);
  });
  let head = 0;
  const order = priorities.sort((a, b) => a - b);
  let count = 0;

  const bfs = () => {
    while (queue.length > head) {
      const [priority, target] = queue[head++];

      if (priority == order[order.length - 1] && target == location)
        return count + 1;

      if (priority != order[order.length - 1]) queue.push([priority, target]);
      else {
        order.pop();
        count++;
      }
    }
  };

  return bfs();
}

File: python/programmers/pg_lv2_프로세스.py
from collections import deque

def solution(priorities, location):
    count = 0
    orderIdx = 0
    sortedPriorities = sorted(priorities, reverse = True)

    def bfs():
        nonlocal count, orderIdx
        queue = deque()
        for (idx, i) in enumerate(priorities):
            queue.append([i, idx])

        while queue:
            num, curIdx = queue.popleft()
            if num == sortedPriorities[orderIdx]:
                if curIdx == location: return count + 1
                else:
                    count += 1
                    orderIdx += 1
            else: queue.append((num, curIdx))

    return bfs()